der/src/asn1/integer/bigint.rs

```diff
@@ -261,6 +265,22 @@ mod allocating {
         }
     }
 
+    impl From<Uint> for Int {
+        fn from(value: Uint) -> Self {
+            let mut inner: Vec<u8> = Vec::new();
+
+            // Add leading `0x00` byte if required
+            if value.value_len().expect("invalid Uint") > value.len() {
+                inner.push(0x00);
+            }
+
+            inner.extend_from_slice(value.as_bytes());
+            let inner = BytesOwned::new(inner).expect("invalid Uint");
+
+            Int { inner }
+        }
+    }
```

Comment:
This provides the infallible `Uint -> Int` conversion, with infallibility under the assumption that the `Uint` was constructed correctly. This is the same as the other `From<...>` implementation directly above, but I can make it `TryFrom` instead if you'd prefer.

Reply:
`From` should be fine here